src/components/CurrentPlaylist.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { TrackItem } from './TrackItem';
+import { ErrorMessage } from './ErrorMessage';
+
+class CurrentPlaylist extends React.Component {
+  state = {
+    error: false,
+    errorMessage: '',
+    tracks: {
+      items: [],
+    },
+    nextTracksEndpoint: null,
+  }
+
+  componentDidMount() {
+    const { token, playlist } = this.props;
+    this.fetchCurrentPlaylistTracks(token, playlist.tracks.href);
+  }
+
+  componentDidUpdate() {
+    const { token } = this.props;
+    const { nextTracksEndpoint } = this.state;
+
+    if (nextTracksEndpoint) {
+      this.fetchCurrentPlaylistTracks(token, nextTracksEndpoint);
+    }
+  }
+
+
+  fetchCurrentPlaylistTracks = (token, endpoint) => {
+    const myHeaders = new Headers();
+    myHeaders.append('Authorization', `Bearer ${token}`);
+
+    fetch(endpoint, {
+      method: 'GET',
+      headers: myHeaders,
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw Error(`Request rejected with status ${response.status}`);
+      })
+      .then((tracks) => {
+        this.setState((prevState) => {
+          const existingTracks = prevState.tracks.items || [];
+          const fetchedTracks = tracks.items;
+          fetchedTracks.forEach((item, i) => { item.key = i + Date.now(); });
+          return {
+            error: false,

```

Comment:
```suggestion
            ...prevState,
            error: false,
```
Usually a good idea to spread prevState so you carry along all previous values and don't stomp on anything during this update.